client/components/wpadmin-auto-login/index.jsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * External Dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { get } from 'lodash';
+
+/**
+ * Try to preload image pixel via going through wp-admin login.
+ * If SSO login pass-thru is enabled, you will be redirected to pixel
+ */
+function tryLogin( requestUrl, initalDelay, attempt ) {
+	const image = new Image();
+	image.src = requestUrl;
+
+	if ( attempt < 9 ) {
+		image.onerror = function() {
+			setTimeout( tryLogin.bind( null, requestUrl, initalDelay, attempt + 1 ), initalDelay * attempt );
+		};
+	}
+}
+
+export default class WpadminAutoLogin extends Component {
+
+	static propTypes = {
+		site: PropTypes.object,
+		delay: PropTypes.number,
+	}
+
+	static defaultProps = {
+		delay: 3000,
+	};
+
+	componentWillMount() {
```

Comment:
How about we update this lifecycle method to `componentDidUpdate`? I can see site being potentially falsey on mount.